### PR TITLE
Fixing HeatTransferFromHeatStructure3D with multiple connected flow channels

### LINKED ISSUE #20298: HeatTransferFromHeatStructure3D1Phase doesn't work with several flow channels
## Bug Description

The component `HeatTransferFromHeatStructure3D1Phase` is not giving physical results when used with several flow channels. 
## Steps to Reproduce

Use this heat transfer component with several flow channels that have different fluid temperatures and look at the resulting heat structure temperature.  It doesn't look physical.

## Impact

Need to use a 1:1 mapping between the heat structure sidesets and the coupled flow channels.

### PR DESCRIPTION
**Pull request by licharlot**
Closes #20298

### CODE REVIEW

File: modules/thermal_hydraulics/src/components/HeatTransferFromHeatStructure3D1Phase.C [hunk 1]
```diff
@@ -43,19 +43,15 @@ HeatTransferFromHeatStructure3D1Phase::HeatTransferFromHeatStructure3D1Phase(
...
     addDependency(fch_name);
   addDependency(_hs_name);
 }
 
-HeatTransferFromHeatStructure3D1Phase::~HeatTransferFromHeatStructure3D1Phase()
-{
-  for (auto & fcha : _fch_alignments)
-    delete fcha;
-}
+HeatTransferFromHeatStructure3D1Phase::~HeatTransferFromHeatStructure3D1Phase() {}
```
Comment (andrsd) [inline modules/thermal_hydraulics/src/components/HeatTransferFromHeatStructure3D1Phase.C:54]: If the d-tor does nothing, then just delete it. The compiler will build the default one for us.

File: modules/thermal_hydraulics/test/tests/components/heat_transfer_from_heat_structure_3d/phy.conservation_ss.i [hunk 2]
```diff
@@ -42,74 +52,123 @@
     initial_p = 1.01e5
     initial_vel = 1
     closures = simple_closures
-    A  = 0.00314159
+    A   = 0.00314159
```
Comment (andrsd) [inline modules/thermal_hydraulics/test/tests/components/heat_transfer_from_heat_structure_3d/phy.conservation_ss.i:55]: Just 1 space after `A`

File: modules/thermal_hydraulics/test/tests/components/heat_transfer_from_heat_structure_3d/phy.conservation_ss.i [hunk 3]
```diff
@@ -42,74 +52,123 @@
...
+    position = '0 0.15 0'
+    orientation = '0 0 1'
+    fp = fp
+    n_elems = 10
+    length = 1
+    initial_T = 350
+    initial_p = 1.01e5
+    initial_vel = 1
+    closures = simple_closures
+    A   = 0.00314159
```
Comment (andrsd) [inline modules/thermal_hydraulics/test/tests/components/heat_transfer_from_heat_structure_3d/phy.conservation_ss.i:81]: Just 1 space after `A`.

**Review by andrsd — CHANGES_REQUESTED** [review]

**Comment by joshuahansel** [comment]
Typo in first commit title and then another in first commit message.

File: modules/thermal_hydraulics/test/tests/components/heat_transfer_from_heat_structure_3d/phy.conservation_ss.i (hunk 2 above)
Comment (licharlot) [reply modules/thermal_hydraulics/test/tests/components/heat_transfer_from_heat_structure_3d/phy.conservation_ss.i]: It was 3 spaces.. Not enough to be noticed by me! :)

**Review by andrsd — APPROVED** [review]

**Pull request merged**